examples/java/src/test/java/org/apache/beam/examples/snippets/SnippetsTest.java

```diff
@@ -154,6 +156,73 @@ public void testCoGroupByKeyTuple() throws IOException {
     p.run();
   }
 
+  /* Tests SchemaJoinPattern */
+  @Test
+  public void testSchemaJoinPattern() {
+    // [START SchemaJoinPatternCreate]
+    // Define Schemas
+    Schema emailSchema =
+        Schema.of(
+            Schema.Field.of("name", Schema.FieldType.STRING),
+            Schema.Field.of("email", Schema.FieldType.STRING));
+
+    Schema phoneSchema =
+        Schema.of(
+            Schema.Field.of("name", Schema.FieldType.STRING),
+            Schema.Field.of("phone", Schema.FieldType.STRING));
+
+    // Create User Data Collections
+    final List<Row> emailUsers =
```

Comment:
Consider if adding a non-match would help the sample. For example a missing id from emailUsers or the other. So that the reader is clear on the output when no match occurs.